pkg/terraform/workspace_test.go

```diff
@@ -0,0 +1,454 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package terraform
+
+import (
+	"bytes"
+	"context"
+	"testing"
+	"time"
+
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/test"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	k8sExec "k8s.io/utils/exec"
+	testingexec "k8s.io/utils/exec/testing"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"github.com/crossplane-contrib/terrajet/pkg/resource/json"
+	tferrors "github.com/crossplane-contrib/terrajet/pkg/terraform/errors"
+)
+
+var (
+	testType              = "very-cool-type"
+	applyType             = "apply"
+	lineage               = "very-cool-lineage"
+	terraformVersion      = "1.0.10"
+	version               = 1
+	serial                = 3
+	directory             = "./testdata"
```

Comment:
We can use `afero`'s in-memory file system for test data, similar to the following test https://github.com/crossplane-contrib/terrajet/blob/8e96bd1/pkg/terraform/files_test.go#L89

Reply:
I noticed that `WorkspaceStore` abstracts the filesystem with `afero` but `Workspace` doesn't. How would you feel about making it use `afero` and use that in your tests?

Reply:
`afero` abstraction was added for `Workspace` and it was used for tests.